### PR TITLE
YG-221: Added an info icon and highlighted the text when the strength is getting full.

### PR DESCRIPTION
**Pull request by Shivani922**
# Related issues
https://pepcus.atlassian.net/browse/YG-221

# Screenshots
image
image
image

### CODE REVIEW

File: src/components/common/ClassroomCard.jsx [hunk 1]
```diff
@@ -30,12 +48,30 @@ const ClassroomCard = ({ isSelected, classroomDetails, classroomNo, onCardSelect
...
-						</StyledCardInfo>
-						<StyledCardInfo>
-							<StyledCardInfoTitle>Current Strength</StyledCardInfoTitle>
-							<StyledCardInfoValue>30</StyledCardInfoValue>
+							<StyledCardInfoValue id="totalStrength">
+								{totalStrength}
+							</StyledCardInfoValue>
 						</StyledCardInfo>
+						{minSeatDifference ? (
+							<StyledCardInfo className="danger-color">
```
Comment (pepcus-mohit) [inline src/components/common/ClassroomCard.jsx:56]: Do not use className on StyledComponents
```suggestion
							<StyledCardInfo currentStrength={currentStrength} totalStrength={totalStrength}>
```

File: src/components/common/ClassroomCard.jsx [hunk 2]
```diff
@@ -117,25 +153,32 @@ const StyledCardInfoContainer = styled.div`
 	border-right: 1px solid #eaeaea;
 	width: 70%;
 	flex-direction: column;
+
+	.danger-color {
```
Comment (pepcus-mohit) [inline src/components/common/ClassroomCard.jsx:157]:
```suggestion
	
```

File: src/components/common/ClassroomCard.jsx [hunk 3]
```diff
@@ -117,25 +153,32 @@ const StyledCardInfoContainer = styled.div`
...
+		color: #e62c14 !important;
```
Comment (pepcus-mohit) [inline src/components/common/ClassroomCard.jsx:158]:
```suggestion
		color: ${totalStrength - currentStrength < = 5 ? 'red' : 'black'};
```

File: src/components/common/ClassroomCard.jsx [hunk 4]
```diff
@@ -1,16 +1,34 @@
 import React, { useState } from 'react';
 import styled from 'styled-components';
 import PropTypes from 'prop-types';
+import Tooltip from 'components/common/Tooltip/Tooltip';
 
 const ClassroomCard = ({ isSelected, classroomDetails, classroomNo, onCardSelection }) => {
 	const [selected, setSelected] = useState();
+	const [minSeatDifference, setMinSeatDifference] = useState(false);
```
Comment (pepcus-mohit) [inline src/components/common/ClassroomCard.jsx:8]: Not Needed
```suggestion
	
```

File: src/components/common/ClassroomCard.jsx [hunk 5]
```diff
@@ -1,16 +1,34 @@
...
+	const [equal, setEqual] = useState(false);
```
Comment (pepcus-mohit) [inline src/components/common/ClassroomCard.jsx:9]: Not needed
```suggestion

```

File: src/components/common/ClassroomCard.jsx [hunk 6]
```diff
@@ -30,12 +48,30 @@ const ClassroomCard = ({ isSelected, classroomDetails, classroomNo, onCardSelect
...
+						{minSeatDifference ? (
+							<StyledCardInfo className="danger-color">
+								<StyledCardInfoTitle>Current Strength</StyledCardInfoTitle>
+								<StyledCardInfoValue>
+									{currentStrength}
+									<Tooltip
+										message={
+											equal
+												? 'No seats left in the classroom '
+												: `Only ${strengthDifference} seats left in the classroom.`
```
Comment (pepcus-mohit) [inline src/components/common/ClassroomCard.jsx:64]: Get these messages from constants in the JSON.

File: src/components/common/ClassroomCard.jsx [hunk 7]
```diff
@@ -30,12 +48,30 @@ const ClassroomCard = ({ isSelected, classroomDetails, classroomNo, onCardSelect
...
+								{totalStrength}
+							</StyledCardInfoValue>
 						</StyledCardInfo>
+						{minSeatDifference ? (
+							<StyledCardInfo className="danger-color">
+								<StyledCardInfoTitle>Current Strength</StyledCardInfoTitle>
+								<StyledCardInfoValue>
+									{currentStrength}
+									<Tooltip
+										message={
```
Comment (pepcus-mohit) [inline src/components/common/ClassroomCard.jsx:61]:
```suggestion
										message={
										totalStrength - currentStrength < = 5 ? ONlY_SEATS_LEFT_MSG : totalStrength - currentStrength <= 0 ? NO_SEATS_LEFT_MSG : null
```

File: src/components/common/ClassroomCard.jsx [hunk 8]
```diff
@@ -1,16 +1,34 @@
...
 	const onChangeCard = (e) => {
 		const target = e.target;
 		if (target.checked) {
 			setSelected(target.value);
 			onCardSelection(classroomDetails);
 		}
 	};
+
+	const strengthDifference = totalStrength - currentStrength;
+	React.useEffect(() => {
```
Comment (pepcus-mohit) [inline src/components/common/ClassroomCard.jsx:23]: Not needed
```suggestion
	
```

**Review by pepcus-mohit — CHANGES_REQUESTED** [review]

File: src/components/common/ClassroomCard.jsx (hunk 1 above)
Comment (Shivani922) [reply src/components/common/ClassroomCard.jsx]: resolved

File: src/components/common/ClassroomCard.jsx (hunk 2 above)
Comment (Shivani922) [reply src/components/common/ClassroomCard.jsx]: Resolved

File: src/components/common/ClassroomCard.jsx (hunk 3 above)
Comment (Shivani922) [reply src/components/common/ClassroomCard.jsx]: Resolved

File: src/components/common/ClassroomCard.jsx (hunk 4 above)
Comment (Shivani922) [reply src/components/common/ClassroomCard.jsx]: Resolved

File: src/components/common/ClassroomCard.jsx (hunk 5 above)
Comment (Shivani922) [reply src/components/common/ClassroomCard.jsx]: Resolved

File: src/components/common/ClassroomCard.jsx (hunk 6 above)
Comment (Shivani922) [reply src/components/common/ClassroomCard.jsx]: Resolved

File: src/components/common/ClassroomCard.jsx (hunk 7 above)
Comment (Shivani922) [reply src/components/common/ClassroomCard.jsx:60]: Resolved

File: src/components/common/ClassroomCard.jsx (hunk 8 above)
Comment (Shivani922) [reply src/components/common/ClassroomCard.jsx]: Resolved

**Review by pepcus-mohit — APPROVED** [review]

**Pull request merged**